### ISSUE TITLE
Custom parameters for XML files

### ISSUE THREAD

**Issue by srounet** [mapnik/mapnik#2738]
Is it possible for a **postgis** datasource to have SQL queries with **external parameters**?

So for example:

```xml
<Datasource>
   <Parameter name="table">
      <![CDATA[(select id, geom FROM my_geometries where c_id =12345) as foo]]>
   </Parameter>
   ...
</Datasource>
```
would become

```xml
<Datasource>
    ...
   <Parameter name="table">
      <![CDATA[(select id, geom FROM my_geometries where c_id = %(c_id)s) as foo]]>
   </Parameter>
   ...
</Datasource>
```

To give a bit more context, I'm trying to move from mapserver to mapnik and as for now I'm using mapserver that way:

1. You have an url where you can add parameters like that: **http://localhost/map/z/x/y.png?c_id=12345**
2. Into mapserver layer configurations you can grab those arguments and inject them into the sql query that way:
```
DATA "geom FROM (
   SELECT id, geom
   FROM my_geometries
   WHERE c_id = '%c_id%'
) as foo using unique id using SRID=4326"
```

Ofc, there are all the stuff around SQL injection etc that mytilecache/mapserver takes care for you, but for the big picture it's like that.

To reformulate my objective here, let's say I have Mapnik + TileStache and I want to control the way some of my layers do generate tiles using custom URL parameters.

As for now, it seems that it's not possible and not even into the Mapnik roadmap, it also seems that it's not even an objective for Mapnik to act like that, and if it's the case I understand (the goal here is different).

But maybe there is a solution for this particular case into custom plugins ?
I've seen that one can write a Plugin using let's say Python Bindings, and maybe implement such a thing.

**Comment by springmeyer** [comment]
Yes, this is supported in Mapnik 3.x (current master and upcoming release). See an example here: https://github.com/mapnik/mapnik/commit/e300a41dca946ddc588339554b7fe376c112101c#diff-82c89228b385f4dbdca7646a3796732bR1014. I would really appreciate help testing the feature since it is new. We do not attempt to sanitize yet to prevent injection to the only safegaurd is read-only transactions (#2142)

**Comment by srounet** [comment]
Is this now supported with the recent release of Mapnik 3.x ?

**Comment by alonek1** [comment]
any update for this issue? is it possible to run sql queries?/
